k8s/resolver_test.py

```diff
@@ -90,6 +90,19 @@ def test_tag_to_digest_sentinel(self):
     }, _BAD_TRANSPORT)
     self.assertEqual(actual_digest, expected_digest)
 
+  def test_tag_embedded_in_metadata(self):
+    """Test that image refs embedded within a YAML string are replaced. This can happen, e.g.,
+    when using spec->template->metadata->annotations.pod.beta.kubernetes.io/init-containers
+    """
+    metadata = '[{"command": ["/some/example/command/example.sh"], "image": "%s"}]'
+    sentinel_string = 'XXXXX'
```

Comment:
Is this `whatever:tag` and thus unique? If so can you change the form to match this?

Otherwise I'm concerned we'll wind up matching something like `bash` which might replace more than just the image.